core/src/components/select/test/fill/select.e2e.ts

```diff
@@ -167,3 +168,60 @@ configs({ modes: ['md'] }).forEach(({ title, screenshot, config }) => {
     });
   });
 });
+
+configs({ modes: ['md'], directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('select: label slot'), () => {
+    test('should render the notch correctly with a slotted label', async ({ page }) => {
+      await page.setContent(
+        `
+        <style>
+          .custom-label {
+            font-size: 30px;
+          }
+        </style>
+        <ion-select
+          fill="outline"
+          label-placement="stacked"
+          value="apple"
+        >
+          <div slot="label" class="custom-label">My Label Content</div>
+          <ion-select-option value="apple">Apple</ion-select-option>
+        </ion-select>
+      `,
+        config
+      );
+
+      const select = page.locator('ion-select');
+      expect(await select.screenshot()).toMatchSnapshot(screenshot(`select-fill-outline-slotted-label`));
+    });
+    test('should render the notch correctly with a slotted label after the select was originally hidden', async ({
+      page,
+    }) => {
+      await page.setContent(
+        `
+        <style>
+          .custom-label {
+            font-size: 30px;
```

Comment:
Setting this style on a floating label causes the label to render too low when the select is closed:

![weird placement](https://i.gyazo.com/c068d910c6312320d6f59c453449cbdb.png)

Not sure if this is best addressed in another PR, though.

Reply:
This is another tricky one. The problem here is we are translating based on %, which will be based on the height of the label element. If the label element grows in height so will the translation.

MDC "fixes" this by giving the label a fixed height. However, this causes the text to get cut off at larger font sizes. This also causes the label to no longer be centered in the container as the select increases in height, though I'm not 100% if that's by design.

We could also make the label height 100%, but then we're going to need to translate by seemingly arbitrary values that may also be wrong as the select changes in height.

Any thoughts on how best to address? I'll keep testing it. It's worth noting that Material Design doesn't seem to do a lot of customizations on floating/stacked labels beyond colors and such. Perhaps this is by design?

Reply:
I was able to get pretty close by changing the following styles on `.label-text-wrapper` (screenshots linked):
- [In the default styles](https://i.gyazo.com/e762efeb8f26a2701a89ea0779e548e5.png), replace the current `transform` with `top: 50%` and `transform: translateY(-50%) scale(1)`. The top: 50/translateY(-50) is a trick I've used to vertically center absolutely positioned items.
- [When floating](https://i.gyazo.com/749fc04809eb7a7f68f6ca9c743b2fea.png), add `top: auto`.

The animation doesn't work right, but maybe there's a good way to fix it? An alternative could be to not do `position: absolute` at all and use flex styles to vertically center the label instead, maybe only when the label isn't floating if the position is necessary to make the float work.

Reply:
During refinement we discussed that this is the same problem as https://ionic-cloud.atlassian.net/browse/FW-4083, so we are going to handle this fix in a separate ticket.